src/Mod/PartDesign/App/FeatureExtrude.cpp

```diff
@@ -860,3 +872,136 @@ App::DocumentObjectExecReturn* FeatureExtrude::buildExtrusion(ExtrudeOptions opt
+                                        true /*CheckUpToFaceLimits.getValue()*/);
+        }
+        catch (Base::Exception&) {
+            if (method == "UpToShape" && faceCount > 1) {
+                throw Base::RuntimeError("Extrude: Unable to reach the selected shape, please select faces");
+            }
+        }
+    }
+    else if (method == "Length" || method == "ThroughAll") {
+        using std::numbers::pi;
+
+        Part::ExtrusionParameters params;
+        params.taperAngleFwd = Base::toRadians(taperAngleDeg);
+
+        if (std::fabs(params.taperAngleFwd) >= Precision::Angular() || std::fabs(params.taperAngleRev) >= Precision::Angular()) {
+            if (fabs(params.taperAngleFwd) > pi * 0.5 - Precision::Angular() || fabs(params.taperAngleRev) > pi * 0.5 - Precision::Angular()) {
+                return prism;
+            }
+            params.dir = dir;
+            params.solid = makeFace;
+            params.lengthFwd = length;
+        
+            std::vector<TopoShape> drafts;
+            Part::ExtrusionHelper::makeElementDraft(params,
+                                                    sketchshape,
+                                                    drafts,
+                                                    getDocument()->getStringHasher());
+            if (drafts.empty()) {
+                return prism;
+            }
+            prism.makeElementCompound(drafts,
+                                      nullptr,
+                                      TopoShape::SingleShapeCompoundCreationPolicy::returnShape);
+        }
+        else {
+            // Without taper angle we create a prism because its shells are in every case no
+            // B-splines and can therefore be use as support for further features like Pads,
+            // Lofts etc. B-spline shells can break certain features, see e.g.
+            // https://forum.freecad.org/viewtopic.php?p=560785#p560785 It is better not to use
+            // BRepFeat_MakePrism here even if we have a support because the resulting shape
+            // creates problems with Pocket
+            try {
+                prism.makeElementPrism(sketchshape, length * gp_Vec(dir));
+            }
+            catch (Standard_Failure&) {
+                throw Base::RuntimeError("FeatureExtrusion: Length: Could not extrude the sketch!");
+            }
+        }
+    }
+
+    return prism;
+}
+
+
+void FeatureExtrude::handleChangedPropertyType(Base::XMLReader& reader,
+                                              const char* TypeName,
+                                              App::Property* prop)
+{
+    // property Type no longer has TwoLengths.
+    if (prop == &Type && strcmp(Type.getValueAsString(), "TwoLengths") == 0) {
+        Type.setValue("Length");
+        Type2.setValue("Length");
+        SideType.setValue("2 sides");
+    }
+    else {
+        ProfileBased::handleChangedPropertyType(reader, TypeName, prop);
+    }
+}
```

Comment:
This technically breaks files for older versions :/ We really need to have some way to handle that gracefully.